uuid-kinds/src/lib.rs

```diff
@@ -20,16 +20,20 @@ use schemars::JsonSchema;
 macro_rules! impl_typed_uuid_kind {
     ($($kind:ident => $tag:literal),* $(,)?) => {
         $(
-            #[cfg_attr(feature = "schemars08", derive(JsonSchema))]
-            pub enum $kind {}
-
-            impl TypedUuidKind for $kind {
-                #[inline]
-                fn tag() -> TypedUuidTag {
-                    // `const` ensures that tags are validated at compile-time.
-                    const TAG: TypedUuidTag = TypedUuidTag::new($tag);
-                    TAG
+            paste::paste! {
```

Comment:
Hmm, `paste` is a proc macro which increases build times, but that's probably ok for this set of things. The alternative would be to have the macro be something like:

```rust
impl_typed_uuid_kind! {
    (DownstairsKind, DownstairsUuid) => "downstairs",
    ...
}
```

which feels worse.

Reply:
While it is a proc macro, shouldn't it only run once for the `omicron-uuid-kinds` crate and then "be done" for other crates which depend on `omicron-uuid-kinds`?

(Very possible I'm wrong on this -- also happy to change it to the version with more arguments if we think the compilation cost would spread)

Reply:
Yes, it's just a one-time thing, and plenty of other crates in omicron have much harsher dependencies on proc macros. I'm just a bit paranoid about proc macro build times, in this case unreasonably so.